`SubStates` derive macro does not work without trait import

## Bevy version and features

- 0.18.0 release or 59e9ee3a
- `features = ["dynamic_linking"]`

## What you did

Tried to start using `SubStates`. Minimal repro:

```rust
#[derive(Clone, Debug, Eq, Hash, PartialEq, bevy::state::state::States)]
enum Foo {
    Foo,
}

#[derive(Clone, Debug, Eq, Hash, PartialEq, Default, bevy::state::state::SubStates)]
#[source(Foo = Foo::Foo)]
enum Bar {
    #[default]
    Bar,
}

fn main() {}
```

## What went wrong

```rust
error[E0599]: no variant or associated item named `SET_DEPENDENCY_DEPTH` found for enum `Foo` in the current scope
 --> src/bin/scratch.rs:6:54
  |
2 | enum Foo {
  | -------- variant or associated item `SET_DEPENDENCY_DEPTH` not found for this enum
...
6 | #[derive(Clone, Debug, Eq, Hash, PartialEq, Default, bevy::state::state::SubStates)]
  |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `Foo`
  |
  = help: items from traits can only be used if the trait is in scope
  = note: this error originates in the derive macro `bevy::state::state::SubStates` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `StateSet` which provides `SET_DEPENDENCY_DEPTH` is implemented but not in scope; perhaps you want to import it
  |
1 + use bevy::prelude::StateSet;
```

The code generated by `derive(SubStates)` appears to depend on having a trait in scope; it should not.

--- a/crates/bevy_state/macros/src/states.rs
+++ b/crates/bevy_state/macros/src/states.rs
@@ -123,9 +123,8 @@ pub fn derive_substates(input: TokenStream) -> TokenStream {
             }
         }
 
-        use #state_set_trait_path as StateSet;
         impl #impl_generics #state_trait_path for #struct_name #ty_generics #where_clause {
-            const DEPENDENCY_DEPTH : usize = <Self as #trait_path>::SourceStates::SET_DEPENDENCY_DEPTH + 1;
+            const DEPENDENCY_DEPTH : usize = <<Self as #trait_path>::SourceStates as #state_set_trait_path>::SET_DEPENDENCY_DEPTH + 1;
         }
 
         impl #impl_generics #state_mutation_trait_path for #struct_name #ty_generics #where_clause {

